iothub/device/src/TransientFaultHandling/ExponentialBackoff.cs

```diff
@@ -98,8 +98,8 @@ public override ShouldRetry GetShouldRetry()
                 if (currentRetryCount < this.retryCount)
                 {
                     Random random = new Random();
-                    int num = (int)((Math.Pow(2.0, (double)currentRetryCount) - 1.0) * (double)random.Next((int)(this.deltaBackoff.TotalMilliseconds * 0.8), (int)(this.deltaBackoff.TotalMilliseconds * 1.2)));
```

Comment:
I don't think this fixes it. Instead of changing the type, I think you need to add `checked { ... }` around the operation.
Please also add a unit test to ensure this will throw on overflow.